cypress/integration/channel/message_permalink.js

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+/* eslint max-nested-callbacks: ["error", 6] */
+
+import users from '../../fixtures/users.json';
+import * as TIMEOUTS from '../../fixtures/timeouts';
+let testChannel;
+let linkText;
+
+function createNewDMChannel(channelname) {
+    cy.get('#addDirectChannel').scrollIntoView().click();
+
+    cy.get('#selectItems').within(() => {
+        cy.get('input[type="text"]').scrollIntoView().type(channelname, {force: true});
+    });
+    cy.contains('.more-modal__description', channelname).click({force: true});
+    cy.get('#saveItems').click();
+}
+
+describe('Message permalink', () => {
+    before(() => {
+        // # Login and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('M13675-Copy a permalink and paste into another channel', () => {
+        const message = 'Hello' + Date.now();
+        const channelName = 'test-message-channel-1';
+
+        // # Create new DM channel with user's email
+        createNewDMChannel(users['user-2'].email);
+
+        // # Post message to user
+        cy.postMessage(message + '{enter}');
+        cy.getLastPostId().then((postId) => {
+            // # check if ... button is visible in last post right side
+            cy.get(`#CENTER_button_${postId}`).should('be.visible');
+
+            // # click on ... button of last post
+            cy.clickPostDotMenu();
+
+            // # click on permalink option
+            cy.contains('Permalink').click();
+
+            // # copy permalink from button
+            cy.get('#linkModalCopyLink').click();
+
+            // # Copy link url from text area
+            cy.get('#linkModalTextArea').invoke('val').then((text) => {
```

Comment:
@saturninoabril  I think we want to use an alias here?

```
cy.get('#linkModalTextArea').invoke('val').as('linkText')
```

Then we'd wrap the section that uses the link text like

```
cy.get('@linkText').then(linkText => {
    // more code
}
```

We could remove the `let linkText` at the top of this file as well then.

Reply:
I couldn't implement  copying content in clipboard and CTRL+ V functionality, as it was not working so i stored the text in separate variable to paste it later. at this point i am not using this text.
[(https://github.com/cypress-io/cypress/issues/2752 )]

Reply:
No need to do anything with copy-paste, IMO. Using the cypress alias is just another way to do variables thats, eh, more "cypress-y". That way we don't have a global-ish variable hanging out in the test (we should also see the variable being called in the test runner, as a bonus). Let me know if that's not clear at all, sorry :)